code/modules/research/xenobiology/crossbreeding/reproductive.dm

```diff
@@ -3,47 +3,31 @@ Reproductive extracts:
 	When fed three monkey cubes, produces between
 	1 and 4 normal slime extracts of the same colour.
 */
+
+
 /obj/item/slimecross/reproductive
 	name = "reproductive extract"
 	desc = "It pulses with a strange hunger."
 	icon_state = "reproductive"
 	effect = "reproductive"
 	effect_desc = "When fed monkey cubes it produces more extracts. Bio bag compatible as well."
 	var/extract_type = /obj/item/slime_extract/
-	var/cubes_eaten = 0
+	var/cooldown = 3 SECONDS
 	var/last_produce = 0
-	var/cooldown = 30 // 3 seconds.
+
+/obj/item/slimecross/reproductive/Initialize()
+	. = ..()
+	LoadComponent(/datum/component/storage/concrete/extract_inventory)
 
 /obj/item/slimecross/reproductive/attackby(obj/item/O, mob/user)
 	if((last_produce + cooldown) > world.time)
 		to_chat(user, span_warning("[src] is still digesting!"))
 		return
 	if(istype(O, /obj/item/storage/bag/bio))
 		var/list/inserted = list()
-		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, 1, null, null, user, inserted)
+		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, 3, null, null, user, inserted)
 		if(inserted.len)
```

Comment:
```suggestion
		if(inserted.len)
            to_chat(user, span_notice("You feed [length(inserted)] Monkey Cube[p_s()] to [src], and it pulses gently."))
            playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
```

you dont want to just try to consume if it inserts, you also want to play the eating sound as well as give the message of how many was inserted